docs/architecture-decisions/0005-implement-an-accessible-navigation-system.md

```diff
@@ -0,0 +1,214 @@
+the supported patterns is the [Disclosure Navigation Menu with Top-Level Links](https://www.w3.org/WAI/ARIA/apg/patterns/disclosure/examples/disclosure-navigation-hybrid/),
+which includes the needed separation between navigation links and submenu
+toggles.
+
+While the provided example includes this separation only at the top level of the
+menu, the pattern can be extended to all levels - [as demonstrated in this
+CodePen](https://codepen.io/matt-bernhardt/pen/poBRKrY).
+
+#### Benefits of Accessible Menu
+
+This menu, when properly configured, appears to implement all the best practices
+for an accessible navigation menu for web application.
+
+#### Drawbacks of Accessible Menu
+
+There are several significant drawbacks to using this menu library.
+
+* The most significant drawback is that it imposes some operating costs on users
+  of assistive technology that are not ideal. The library forces the screen
+  reader user to explicitly switch operating mode to interact with the menu, and
+  does not listen for the type of interaction events which users naturally
+  generate while navigating a page.
+
+  > However, the library is not listening for click events; it seems they are
+  > only listening for explicit pointer events. This means you need to switch
+  > into forms mode, then press enter or space, to activate a menu ... It also
+  > seems they are handling keyboard events explicitly for keys that should
+  > normally be handled by the browser like tab and shift+tab.
+
+  (The above comment comes from [our discussion of the navigation menu after
+  accessibility testing](https://github.mit.edu/Accessibility-FY2024/omeka/issues/4#issuecomment-182190).)
+
+* The markup requirements for this library do not immediately align with the
+  markup generated by Omeka. While this is manageable by either maintaining a
+  set of custom navigation templates (which other themes also do) or by writing
+  some javascript to manipulate the DOM prior to instantiating the menu, it does
+  mean that using this library increases the maintenance burden of the theme.
+
+* The library is javascript-only, and is not distributed with any CSS rules.
+  While we can develop and maintain these rules ourselves, it makes this library
+  less of a turnkey solution.
+
+
+### Option 3: CSS-only menu libraries
+
+A third option would be to pursue a "pure CSS" navigation menu, which can be
+found described in [several](https://purecss.io/menus/) [places](https://devsnap.me/css-menu) around the web.
+
+* Pros: A navigation menu which uses only CSS styles is technically much
+  simpler, lacking the javascript that would need to be maintained (either by
+  ourselves or by a dependency maintainer).
+
+* Cons: [These types of navigation systems are likely not going to be fully
+  accessible](https://moderncss.dev/css-only-accessible-dropdown-navigation-menu/). There are elements of the WCAG standards which cannot be satisfied
```

Comment:
This feels like a dealbreaker (which is unfortunate, because a CSS-only menu is strangely compelling to me).

Reply:
Agreed, on both points.